pkg/util/tracing/collector/collector_test: TestClusterInflightTraces failed

pkg/util/tracing/collector/collector_test.TestClusterInflightTraces [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10939831?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10939831?buildTab=artifacts#/) on release-23.1 @ [5c96635393a0fa869fc478225756c21ca994e079](https://github.com/cockroachdb/cockroach/commits/5c96635393a0fa869fc478225756c21ca994e079):


```
=== RUN   TestClusterInflightTraces
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f694b965ca502737ec1f9c8071e6a7a2/logTestClusterInflightTraces2281702773
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestClusterInflightTraces
    collector_test.go:363: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f694b965ca502737ec1f9c8071e6a7a2/logTestClusterInflightTraces2281702773
--- FAIL: TestClusterInflightTraces (30.76s)
=== RUN   TestClusterInflightTraces/shared-process
    collector_test.go:274: 
        	Error Trace:	pkg/util/tracing/collector/collector_test_test/pkg/util/tracing/collector/collector_test.go:274
        	Error:      	Received unexpected error:
        	            	server stop before successful start
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*channelOrchestrator).startControlledServer.func5.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_channel_orchestrator.go:307
        	            	  | github.com/cockroachdb/cockroach/pkg/server.(*channelOrchestrator).startControlledServer.func5
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/server_controller_channel_orchestrator.go:401
        	            	  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) server stop before successful start
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestClusterInflightTraces/shared-process
    --- FAIL: TestClusterInflightTraces/shared-process (7.49s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterInflightTraces.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29756

--- a/pkg/util/tracing/collector/collector_test.go
+++ b/pkg/util/tracing/collector/collector_test.go
@@ -270,9 +270,6 @@ func TestClusterInflightTraces(t *testing.T) {
 				for i, s := range tc.Servers {
 					tenant, db, err := s.StartSharedProcessTenant(ctx, base.TestSharedProcessTenantArgs{TenantName: "app"})
 					require.NoError(t, err)
-					defer func() {
-						require.NoError(t, db.Close())
-					}()
 					tenants[i] = tenant
 					dbs[i] = db
 				}

